server/src/main/java/io/whitefox/core/services/DeltaSharesServiceImpl.java

```diff
@@ -4,13 +4,13 @@
 import io.whitefox.core.Share;
 import io.whitefox.core.Table;
 import io.whitefox.persistence.StorageManager;
-import jakarta.enterprise.context.ApplicationScoped;
 import jakarta.inject.Inject;
+import jakarta.inject.Singleton;
 import java.util.List;
 import java.util.Optional;
 import org.eclipse.microprofile.config.inject.ConfigProperty;
 
-@ApplicationScoped
```

Comment:
why do we move those from appscope to singleton? 
if I understand well from SO -> no serialization needed(so we dont need app scope), and eagerly instantiated

Reply:
looks at my comment :) 
https://github.com/agile-lab-dev/whitefox/pull/72#pullrequestreview-1675913695

Reply:
can we mock an api with singleton anyhow or is it just not possible with quarkusMock?
is there a way to measure coverage using scoped? if its possible I would stick with it, since I dont see that much benefits from the singleton besides being able to run coverage, even with the client proxy it seems fast enough for our usecase(going by the logic of removing async)

Reply:
yeah, I'm pissed about the coverage mainly, other reasons are there because I know of them, but are not a real problem

Reply:
Basically I don't like Singleton, since it's not mockable. But in this case, at least for now, we don't have to mock it since we are creating delta tables locally for the IT-Test. Also, we get the coverage with jacoco. 

If in the future we understand how to fix jacoco on application scoped classes, I think we can remove the singleton annotation.

Reply:
We can't use it in scenarios like this:
https://github.com/agile-lab-dev/whitefox/blob/11d9755cfc3875a8a99d4fe1c1420998c28703b9/server/src/test/java/io/whitefox/api/deltasharing/server/DeltaSharesApiImplTest.java#L41